tests/test_signed_xpi.py

```diff
@@ -56,3 +56,73 @@ def test_other_signed_xpi():
     assert not err.errors
 
     eq_(err.warnings[0]['id'], ('testcases_content', 'signed_xpi'))
+
+
+def test_mozilla_signed_multi_xpi():
+    """Test that mozilla signed inner packages don't raise an error."""
+    xpi = MockXPI({
+        'META-INF/manifest.mf': 'tests/resources/main/foo.bar',
+        'META-INF/mozilla.rsa': 'tests/resources/main/foo.bar',
+        'META-INF/mozilla.sf': 'tests/resources/main/foo.bar',
+    }, subpackage=True)
+
+    err = ErrorBundle()
+    err.supported_versions = {}
+    err.push_state('sub.xpi')
+    err.save_resource('is_multipackage', True)
+
+    content.test_signed_xpi(err, xpi)
+
+    assert not err.failed()
+
+
+def test_unsigned_multi_xpi():
+    """Test that unsigned inner packages raise an error."""
+    xpi = MockXPI(subpackage=True)
+
+    err = ErrorBundle()
+    err.supported_versions = {}
+    err.push_state('sub.xpi')
+    err.save_resource('is_multipackage', True)
+
+    content.test_signed_xpi(err, xpi)
+
+    assert err.failed()
+    assert not err.warnings
+    assert err.errors
+    eq_(err.errors[0]['id'], ('testcases_content', 'unsigned_sub_xpi'))
+
+
+def test_other_signed_multi_xpi():
+    """Test that other signed inner packages raise an error."""
+    xpi = MockXPI({
+        'META-INF/manifest.mf': 'tests/resources/main/foo.bar',
+        'META-INF/zigbert.rsa': 'tests/resources/main/foo.bar',
+        'META-INF/zigbert.sf': 'tests/resources/main/foo.bar',
+    }, subpackage=True)
+
+    err = ErrorBundle()
+    err.supported_versions = {}
+    err.push_state('sub.xpi')
+    err.save_resource('is_multipackage', True)
+
+    content.test_signed_xpi(err, xpi)
+
+    assert err.failed()
+    assert not err.warnings
+    assert err.errors
+    eq_(err.errors[0]['id'], ('testcases_content', 'unsigned_sub_xpi'))
+
+
+def test_unsigned_inner_xpi():
+    """Test that inner packages don't need to be signed if not multipackage."""
```

Comment:
There's a jar file in test-amo-validator that has `is_nested_package` but isn't a multi package. I'm not quite sure why but maybe because the jar file can't be the root package?